mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/mosaic.kt

```diff
@@ -106,16 +113,22 @@ public suspend fun runMosaic(content: @Composable () -> Unit): Unit = coroutineS
 	val rootNode = createRootNode()
 	val applier = MosaicNodeApplier(rootNode) {
 		val render = rendering.render(rootNode)
-		platformDisplay("$ansiBeginSynchronizedUpdate$render$ansiEndSynchronizedUpdate")
+		display("$ansiBeginSynchronizedUpdate$render$ansiEndSynchronizedUpdate")
 	}
-
+	val composition = Composition(applier, recomposer)
 	try {
-		Composition(applier, recomposer).setContent {
+		composition.setContent {
 			CompositionLocalProvider(LocalTerminal provides terminalInfo.value) {
 				content()
 			}
 		}
 
+		val effectJob = checkNotNull(recomposer.effectCoroutineContext[Job]) {
+			"No Job in effectCoroutineContext of recomposer"
+		}
+		effectJob.children.forEach { it.join() }
+		recomposer.awaitIdle()
```

Comment:
It turned out to be necessary to pass the `counter` unit test, without this, the last render is not done with `TestDispatcher`s. If you specify `Dispatchers.Default` or make a `delay` after assigning a value to `count`, it will work, but it doesn't seem like it's super.